lib/src/windows/file_picker_windows.dart

```diff
@@ -94,78 +95,94 @@ class FilePickerWindows extends FilePicker {
+
     int hr = CoInitializeEx(
       nullptr,
       COINIT.COINIT_APARTMENTTHREADED | COINIT.COINIT_DISABLE_OLE1DDE,
     );
 
-    // Ignore the error if COM is already initialized.
-    // If this is the case, CoInitializeEx will return
-    // RPC_E_CHANGED_MODE.
-    if (!SUCCEEDED(hr) && hr != RPC_E_CHANGED_MODE) {
+    if (!SUCCEEDED(hr)) {
       throw WindowsException(hr);
     }
 
     final fileDialog = FileOpenDialog.createInstance();
 
     final optionsPointer = calloc<Uint32>();
-    hr = fileDialog.getOptions(optionsPointer);
-    if (!SUCCEEDED(hr)) throw WindowsException(hr);
+    try {
+      hr = fileDialog.getOptions(optionsPointer);
+      if (!SUCCEEDED(hr)) throw WindowsException(hr);
 
-    final options = optionsPointer.value |
-        FILEOPENDIALOGOPTIONS.FOS_PICKFOLDERS |
-        FILEOPENDIALOGOPTIONS.FOS_FORCEFILESYSTEM |
-        FILEOPENDIALOGOPTIONS.FOS_NOCHANGEDIR;
-    hr = fileDialog.setOptions(options);
-    if (!SUCCEEDED(hr)) throw WindowsException(hr);
+      final options = optionsPointer.value |
+          FILEOPENDIALOGOPTIONS.FOS_PICKFOLDERS |
+          FILEOPENDIALOGOPTIONS.FOS_FORCEFILESYSTEM |
+          FILEOPENDIALOGOPTIONS.FOS_NOCHANGEDIR;
+      hr = fileDialog.setOptions(options);
+      if (!SUCCEEDED(hr)) throw WindowsException(hr);
+    } finally {
+      free(optionsPointer);
+    }
 
-    final title = TEXT(dialogTitle ?? defaultDialogTitle);
-    hr = fileDialog.setTitle(title);
-    if (!SUCCEEDED(hr)) throw WindowsException(hr);
-    free(title);
+    final title = TEXT(dialogTitle ?? 'Select a Folder');
+    try {
+      hr = fileDialog.setTitle(title);
+      if (!SUCCEEDED(hr)) throw WindowsException(hr);
+    } finally {
+      free(title);
+    }
 
     if (initialDirectory != null) {
       final folder = TEXT(initialDirectory);
       final riid = convertToIID(IID_IShellItem);
       final ppv = calloc<Pointer>();
-      hr = SHCreateItemFromParsingName(folder, nullptr, riid, ppv);
-      final item = IShellItem(ppv.cast());
-      free(riid);
-      free(folder);
-      if (!SUCCEEDED(hr)) throw WindowsException(hr);
-      hr = fileDialog.setFolder(item.ptr.cast<Pointer<COMObject>>().value);
-      if (!SUCCEEDED(hr)) throw WindowsException(hr);
-    }
 
-    final hwndOwner = lockParentWindow ? GetForegroundWindow() : NULL;
-    hr = fileDialog.show(hwndOwner);
-    if (!SUCCEEDED(hr)) {
-      CoUninitialize();
+      try {
+        hr = SHCreateItemFromParsingName(folder, nullptr, riid, ppv);
+        if (!SUCCEEDED(hr) || ppv.value == nullptr) {
+          throw WindowsException(hr);
+        }
 
-      if (hr == HRESULT_FROM_WIN32(WIN32_ERROR.ERROR_CANCELLED)) {
-        return Future.value(null);
+        final item = IShellItem(ppv.cast());
+        hr = fileDialog.setFolder(item.ptr.cast<Pointer<COMObject>>().value);
+        if (!SUCCEEDED(hr)) throw WindowsException(hr);
+      } finally {
+        free(riid);
```

Comment:
Nit: can we sort these `free()` calls  in the order that the pointers are declared?